odata_query/django/django_q.py

```diff
@@ -1,14 +1,36 @@
 import operator
-from typing import Any, Callable, Dict, List, Optional, Type
-
-from django.db.models import Exists, F, Model, OuterRef, Q, Value, functions
+from typing import Any, Callable, Dict, List, Optional, Type, Union
+from uuid import UUID
+
+import django
+from django.db.models import (
+    Case,
+    Exists,
+    F,
+    Model,
+    OuterRef,
+    Q,
+    Value,
+    When,
+    functions,
+    lookups,
+)
 from django.db.models.expressions import Expression
 
 from odata_query import ast, exceptions as ex, typing, utils, visitor
 
+from .django_q_ext import NotEqual
 from .utils import reverse_relationship
 
+DJANGO_LT_4 = django.VERSION[0] < 4
+
 COMPARISON_INVERT = {
```

Comment:
This is just a small thing, but I first interpreted “invert” as “negate” here, leading to slight confusion when reading the items below. What do you think of naming this `COMPARISON_FLIP`, analogous to the method that uses it?

Reply:
Great point!